src/applications/personalization/profile-2/components/ProfileHeader.jsx

```diff
@@ -1,19 +1,85 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
-const ProfileHeader = () => (
-  <div className="vads-u-background-color--primary-darkest vads-u-color--white vads-u-margin-bottom--2 vads-u-padding-y--3">
-    <div className="row">
-      <h1 className="vads-u-font-family--sans vads-u-font-size--base vads-u-font-weight--normal">
-        Your Profile
-      </h1>
-      <h2 className="vads-u-font-size--h3">
-        Kimberly Elizabeth Smith Washington
-      </h2>
-      <h3 className="vads-u-font-family--sans vads-u-font-size--base vads-u-font-weight--normal">
-        United States Army Reserve
-      </h3>
+const ProfileHeader = ({ userFullName, latestBranchOfService }) => {
+  const imagePaths = {
+    A: '/img/vic-army-symbol.png',
+    C: '/img/vic-cg-emblem.png',
+    F: '/img/vic-air-force-coat-of-arms.png',
+    N: '/img/vic-navy-emblem.png',
+    M: '/img/vic-usmc-emblem.png',
+    VASeal: '/img/vic-va-seal.png',
+  };
+
+  const modifyNameWith = (name, addition = '') => {
+    if (!name) {
+      return addition;
+    }
+    return `${name} ${addition}`;
+  };
+
+  const renderName = ({ first, middle, last }) => {
+    let name = '';
+    // Attempt to compile first + middle + last.
+    name = modifyNameWith(name, first);
+    name = modifyNameWith(name, middle);
+    name = modifyNameWith(name, last);
+    return name;
```

Comment:
Oh good idea! I think you'll have to trim it, but this would work:
```
[undefined, 'hello', 'there'].join(' ') // " hello there"
```

You could use [_.trim](https://lodash.com/docs/4.17.15#trim) to remove the leading/trailing whitespace! 🙂

Reply:
That won’t handle a missing middle name. Maybe put them in an array, filter out anything that isn’t a string with a length, and join that.

Reply:
The whitespace will be reduced once in the HTML anyway, right?

Reply:
We may also want to validate the first name -> 

```js
if (!['john', 'erik', 'kelson', 'brad', 'nick', /* etc */ ].includes(first)) 
    alert('your first name is not valid')
```

Reply:
hahahaha `alert` is my favorite user experience

Reply:
At any rate, the code you wrote here @bradl3yC works so in the end it's your call! 🙂

Reply:
@erikphansen is correct - ```const fullName = [first, middle, last].filter(name => !!name).join(' ');``` works great